dnsext-types/DNS/Types/Domain.hs

```diff
@@ -182,25 +183,38 @@ modifyMailbox f (Mailbox o l) = Mailbox (f o) (f l)
 
 ----------------------------------------------------------------
 
-putDomain :: Domain -> SPut
-putDomain (Domain o _) = putDomain' _period o
+-- | Canonical-form flag.
+-- For example, `Canonical` is used from DNSSEC extension.
+--
+-- ref. https://datatracker.ietf.org/doc/html/rfc4034#section-6.2 - Canonical RR Form
+data CanonicalFlag
+  = NoCanonical
```

Comment:
`Compression` would be better than `NoCanonical`.

Reply:
https://github.com/kazu-yamamoto/dnsext/pull/13/commits/3bfc0d3c9b6a2ac5623f0ccf74b5eb8ad435594d